server/src/main/java/org/elasticsearch/action/admin/indices/shrink/ResizeRequest.java

```diff
@@ -102,6 +117,13 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(sourceIndex);
         out.writeEnum(type);
         out.writeOptionalBoolean(copySettings);
+        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+            boolean hasMaxSinglePrimarySize = maxSinglePrimarySize != null;
+            out.writeBoolean(hasMaxSinglePrimarySize);
+            if (hasMaxSinglePrimarySize) {
+                maxSinglePrimarySize.writeTo(out);
+            }
```

Comment:
This can be simplified to
```suggestion
            out.writeOptionalWriteable(maxSinglePrimarySize);
```